Source/Core/Common/File.h

```diff
@@ -67,6 +68,10 @@ class IOFile
     return WriteArray(reinterpret_cast<const char*>(data), length);
   }
 
+  bool WriteString(const char* str) { return WriteString(std::string_view(str)); }
+  bool WriteString(const std::string_view& str) { return WriteBytes(str.data(), str.size()); }
```

Comment:
```suggestion
  bool WriteString(std::string_view str) { return WriteBytes(str.data(), str.size()); }
```

This overload alone should be sufficient to make the const char* and std::string overloads unnecessary as well.

Reply:
I was under the impression that that had weird edge cases where the string could be deallocated before the string_view is used, but trying to produce such a case was fruitless, so yeah I guess that works!

Reply:
That is only a problem if we keep the string_view around after the function finishes executing. (For instance, if we return it from the function.)